### PR TITLE
Gfx1030 supported fmac

### PR DESCRIPTION
**Pull request by ltqin**
Solve the issue #484
change 2 items:
1.Replace v_mac_f32 as v_fmac_f32, if flag CK_USE_AMD_V_FMAC_F32 is true
2.Disable buffer_load/store when device name is gfx1030

### CODE REVIEW

**Comment by atamazov** [comment]
[off-topic] @ltqin There is no need to insert full URL links when hyperlinking whithin the same repo. Just typing `#484` is enough.

File: src/kernels/composable_kernel/include/utility/amd_inline_asm.hpp [hunk 1]
```diff
@@ -9,18 +9,37 @@ namespace ck {
 // outer-product: c[i,j] += inner_product(a[i], b[j])
 __device__ void amd_assembly_outer_product_1x2(float a, float b0, float b1, float& c0, float& c1)
 {
+#if CK_USE_AMD_V_FMAC_F32
```
Comment (asroy) [inline src/kernels/composable_kernel/include/utility/amd_inline_asm.hpp:12]: Why not removing v_mac_f32 completely?

**Review by asroy — CHANGES_REQUESTED** [review]

**Review by atamazov — DISMISSED** [review]

File: src/solver/implicitgemm_util.hpp [hunk 2]
```diff
@@ -707,6 +707,22 @@ static inline bool support_amd_buffer_atomic_fadd(const ConvolutionContext& ctx)
     return StartsWith(device_name, "gfx908");
 }
 
+static inline bool device_name_is_gfx1030(const ConvolutionContext& ctx)
+{
+    const auto device_name = ctx.GetStream().GetDeviceName();
+    return StartsWith(device_name, "gfx1030");
+}
+
+static inline bool support_amd_buffer_load_store(const ConvolutionContext& ctx)
```
[the remark this answers is not in the document]
Comment (asroy) [reply src/solver/implicitgemm_util.hpp:716]: Right. gfx1030 support buffer instructions. But the llvm intrinsic functions we are using are deprecated by compiler team and we need to be updated. So yea, it's a workaround for now.

File: src/kernels/composable_kernel/include/utility/amd_inline_asm.hpp (hunk 1 above)
Comment (atamazov) [reply src/kernels/composable_kernel/include/utility/amd_inline_asm.hpp:12]: @asroy 
- gfx900 foes not have FMAC_F32
- MAC_F32 -> FMAC_F32 may lead to performance drop on gfx9. More info upon request.

File: src/solver/implicitgemm_util.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (ltqin) [reply src/solver/implicitgemm_util.hpp:716]: @atamazov I have not right to access AMDComputeLibraries/MLOpen#2331

[the remark this answers is not in the document]
Comment (atamazov) [reply src/solver/implicitgemm_util.hpp]: See #561.

Please also ask @daniellowell to provide you with access to private repo.

File: src/kernels/composable_kernel/include/utility/config.hpp [hunk 3]
```diff
@@ -38,6 +38,11 @@
 #define CK_USE_AMD_BUFFER_ATOMIC_FADD 0
 #endif
 
+// gfx1030 only use v_fmac_f32
```
[the remark this answers is not in the document]
Comment (ltqin) [reply src/kernels/composable_kernel/include/utility/config.hpp]: fixed

File: src/solver/implicitgemm_util.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (ltqin) [reply src/solver/implicitgemm_util.hpp]: add WORKAROUND_MIOPEN_ISSUE_557 for this

File: src/solver/implicitgemm_util.hpp [hunk 4]
```diff
@@ -801,6 +817,14 @@ static inline auto get_ck_common_compiler_flag(const ConvolutionContext& ctx)
         std::string(" -DCK_WORKAROUND_SWDEV_229564=") + std::to_string(WORKAROUND_SWDEV_229564) +
         std::string(" -DCK_WORKAROUND_SWDEV_231101=") + std::to_string(WORKAROUND_SWDEV_231101);
 
+    // fgx1030 disable buffer load/store
```
[the remark this answers is not in the document]
Comment (ltqin) [reply src/solver/implicitgemm_util.hpp]: fixed

**Review by asroy — CHANGES_REQUESTED** [review]

File: src/solver/implicitgemm_util.hpp [hunk 5]
```diff
@@ -707,6 +707,22 @@ static inline bool support_amd_buffer_atomic_fadd(const ConvolutionContext& ctx)
...
 
+static inline bool device_name_is_gfx1030(const ConvolutionContext& ctx)
+{
+    const auto device_name = ctx.GetStream().GetDeviceName();
+    return StartsWith(device_name, "gfx1030");
+}
+
+static inline bool is_use_amd_buffer_load_store(const ConvolutionContext& ctx)
+{
+    return !device_name_is_gfx1030(ctx);
```
[the remark this answers is not in the document]
Comment (atamazov) [reply src/solver/implicitgemm_util.hpp:723]: @asroy I think we already have issue #557 and this PR introduces relevant macro (`WORKAROUND_MIOPEN_ISSUE_557`).

[the remark this answers is not in the document]
Comment (ltqin) [reply src/solver/implicitgemm_util.hpp:723]: @asroy Did you mean I should remove function of device_name_is_gfx1030

**Comment by atamazov** [comment]
@ltqin Please assign at least one reviewer to this PR.

**Comment by rocm-mici** [comment]
Status?

File: src/solver/implicitgemm_util.hpp [hunk 6]
```diff
@@ -707,6 +707,22 @@ static inline bool support_amd_buffer_atomic_fadd(const ConvolutionContext& ctx)
...
+}
+
+static inline bool is_use_v_fmac_f32(const ConvolutionContext& ctx)
+{
+    return device_name_is_gfx1030(ctx);
```
[the remark this answers is not in the document]
Comment (ltqin) [reply src/solver/implicitgemm_util.hpp:731]: I think  writing a function is more readable

**Review by asroy — APPROVED** [review]

**Pull request merged**